fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -215,6 +231,7 @@ where
             .get_top_down_msgs(height)
             .await
             .map_err(|e| Error::CannotQueryParent(e.to_string(), height))?;
+
```

Comment:
if we detected a chain reorg, do we have to clear existing votes?

Reply:
Thanks for highlighting this; the tally doesn't handle reorgs in any way at the moment. I wasn't sure how well it worked with this component either, so I ignored it. I thought it might be a remnant of when we fetched up to the tip of the chain where it was volatile, but as I understand currently it only fetches what is supposed to be final, and then it votes for that too, so if everyone votes it gets finalized and no reorg can happen, or at least we can't handle it.